components/angular-tomitribe-tags/src/tags.controller.ts

```diff
@@ -23,13 +25,41 @@ export class TagsController {
 
         $scope.$on('tribe-tags:refresh', () => $scope.loadTagsProposals(''));
 
-        $scope.loadTagsProposals = (query) => tagService.findTags(query).then(
-            (data) => {
-                $scope.availableTags = data.map(t => {
-                    return new TagReference(t.id, t.name, {});
-                })
-                .sort((a,b) => a.name.localeCompare(b.name));
+        $scope.loadTagsProposals = (query) => {
+            if ($scope.$$pagingBusy) return;
+
+            if (query !== $scope.$$pagingQuery) {
+                $scope.$$pagingQuery = query;
+                $scope.$$pagingState = undefined;
             }
-        );
+
+            // we started our request (busy)
+            $scope.$$pagingBusy = true;
+
+            let params = {
+                "pagingState": $scope.$$pagingState || undefined,
+                "query": query || undefined,
+                "size": 20
+            };
+
+            tagService.findTags(params).then(
+                (data) => {
+                    let tagsFromServer = data.items.map(t => {
+                        return new TagReference(t.id, t.name, {});
+                    });
+
+                    const previousArray = $scope.availableTags && $scope.$$pagingState ? $scope.availableTags : [];
+
+                    $scope.availableTags = _.union(previousArray, tagsFromServer).sort((a:any,b:any) => a.name.localeCompare(b.name));
```

Comment:
If possible please use same method as in TAG-2476, I suggest adding it to one of components *(select for example or ask andrii.p to do that)